constraint/chip_analysis.py

```diff
@@ -104,71 +36,210 @@ def rename_cols(column):
-df[['ks_statistic_ab','ks_p_value_ab']] = df[['expanded_hists_ab_lof','expanded_hists_ab_synonymous']].apply(lambda x: test_ks(*x), axis=1)
-df[['ks_statistic_age','ks_p_value_age']] = df[['expanded_hists_age_lof','expanded_hists_age_synonymous']].apply(lambda x: test_ks(*x), axis=1)
-
-# test lof vs synonymous distributions for ab and age using the median histogram data
-expanded_hist_ab_median = expand_hist(median_ab_hist,bin_floors_ab)
-expanded_hist_age_median = expand_hist(median_age_hist,bin_floors_age)
-
-df[['ks_statistic_ab_median','ks_p_value_ab_median']] = (df['expanded_hists_ab_lof']).apply(test_ks, args=[expanded_hist_ab_median])
-df[['ks_statistic_age_median','ks_p_value_age_median']] = (df['expanded_hists_age_lof']).apply(test_ks, args=[expanded_hist_age_median])
-
-# sort by p-value for ks test on ab
-df = df.sort_values(by=['ks_p_value_ab'])
-
-# get lengths of arrays that were used for ks test
-df['len_expanded_hists_ab_lof'] = df['expanded_hists_ab_lof'].apply(lambda x: len(x) if isinstance(x,np.ndarray) else np.NaN)
-df['len_expanded_hists_ab_synonymous'] = df['expanded_hists_ab_synonymous'].apply(lambda x: len(x) if isinstance(x,np.ndarray) else np.NaN)
-df['len_expanded_hists_age_lof'] = df['expanded_hists_age_lof'].apply(lambda x: len(x) if isinstance(x,np.ndarray) else np.NaN)
-df['len_expanded_hists_age_synonymous'] = df['expanded_hists_age_synonymous'].apply(lambda x:  len(x) if isinstance(x,np.ndarray) else np.NaN)
-df['len_expanded_hist_ab_median'] = len(expanded_hist_ab_median)
-df['len_expanded_hist_age_median'] = len(expanded_hist_age_median)
-
-
-# retain only certain columns
-df_out = df[['transcript_id',
-             'gene_symbol',
-             'summed_ab_hists_lof',
-             'summed_age_hists_lof',
-             'summed_ab_hists_synonymous',
-             'summed_age_hists_synonymous',
-             'ks_statistic_ab',
-             'ks_p_value_ab',
-             'ks_statistic_age',
-             'ks_p_value_age',
-             'ks_statistic_ab_median',
-             'ks_p_value_ab_median',
-             'ks_statistic_age_median',
-             'ks_p_value_age_median',
-             'len_expanded_hists_ab_lof',
-             'len_expanded_hists_ab_synonymous',
-             'len_expanded_hists_age_lof',
-             'len_expanded_hists_age_synonymous',
-             'len_expanded_hist_ab_median',
-             'len_expanded_hist_age_median']]
+    
+    
+def test_mood(a1,a2):
+    """
+    Runs the two sample Kolmogorov-Smirnov test on the two supplied arrays, requires ndarrays with at least 2 distinct values
+    :param list a1: array 1
+    :param list a2: array 2
+    :return: [KS statisic, KS p-value]
+    :rtype: list
+    """
+    if isinstance(a1,np.ndarray) and isinstance(a2,np.ndarray):
+        if len(set(a1)) > 1 and len(set(a2)) > 1:  # require at least 2 distinct values
+            stat,p,grand_median,contigency_table = stats.median_test(a1,a2)
```

Comment:
spaces after splitting a tuple out: `stat, p, ...`, but also, if you only ever use `p` you could either do: `_, p, _, _ = stats.median_test` or `values = stats.median_test` and then `return values[1]`. but this is fine